app/jobs/workoff_arbiter.rb

```diff
@@ -61,12 +61,40 @@ def add_worker!
       ],
     }
 
-    ecs.run_task(payload)
-    @notifier.ping("Added a workoff worker. Metric was #{metric.round} (#{_dj_scope.pluck(:id).count} jobs enqueued) with #{_current_worker_count} workers right now (this might include the just-created one).")
+    _num_workers_to_add.times do
+      ecs.run_task(payload)
+      @notifier.ping("Added a workoff worker. Long jobs pending: #{_long_jobs.count}. Metric was #{metric.round} (#{_dj_scope.pluck(:id).count} jobs enqueued) with #{_current_worker_count} workers right now (this might include the just-created one).")
+    end
   end
 
   private
 
+  def _long_work_pending?
+    _long_jobs.any?
+  end
+
+  def _num_workers_to_add
+    if _long_work_pending?
+      [_long_jobs.count, 3].min
+    else
+      1
+    end
+  end
+
+  def _long_jobs
+    @_long_jobs ||=
+      begin
+        # This prevents this code from silently failing to work correctly if we
+        # were to change the queue name in the future
+        queues = Delayed::Worker.queue_attributes.keys
+        long_queue = 'long_running'
+        raise "Incorrect queue: #{long_queue}" unless long_queue.in?(queues)
+
+        Delayed::Job.where(queue: long_queue, locked_by: nil, last_error: nil, failed_at: nil)
```

Comment:
I'm tempted to invert this to `where.not(queue: 'short_running')` Or maybe drop the queue check entirely, just spool them up if we have jobs in the queue that aren't being worked on.

Reply:
Let's ignore the `mailers` queue when determining if we should spin up a worker.  The short running should handle mailers.

Arbiter class will determine length of queue as any unstarted jobs in any queue except `mailers`.